### PR TITLE
docs: improve `DataFrame`, `LazyFrame` and `Series` docstrings

### LINKED ISSUE #1618: [Enh]: Adding a series as a new column to a dataframe
Say I have a list of values and I want to create a new column using that list of values. In polars this can be done:
```

import polars as pl

df = pl.DataFrame(
    {"player": ["a", "b", "a", "c", "b", "a"], "value": [10, 20, 15, 5, 30, 25]}
)

list_a = [1, 0, 1, 0, 0, 1]

new_df = df.with_columns(
            pl.Series(name="new_column",values=list_a)
        )
```

However, I am struggling to figure out how to do this in Narwhals. The nw.Series works differently. Reading the documentation I see     

> This class is not meant to be instantiated directly - instead, use
>     `narwhals.from_native`, making sure to pass `allow_series=True` or
>     `series_only=True`.

But I am not exactly sure how to use that information to solve my issue. 

### Suggest a solution if possible.

I think it would be nice if it could work similarly to polars: 

```
import narwhals as nw
from narwhals.typing import FrameT

@nw.narwhalify
def add_np_array(df: FrameT) -> FrameT:
    return df.with_columns(
            nw.Series("new_column",[1, 0, 1, 0, 0, 1])
        )

new_df = add_np_array(df)
```
Although I appreciate any suggestion on how to solve my issue.

### PR DESCRIPTION
**Pull request by FBruzzesi**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [x] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues

- Closes #1618

## Checklist

- [ ] Code follows style guide (ruff)
- [ ] Tests added
- [ ] Documented the changes

## If you have comments or can explain your changes, please do so below

Example rendering

image

### CODE REVIEW

File: narwhals/expr.py [hunk 1]
```diff
@@ -6087,7 +6087,7 @@ def nth(*indices: int | Sequence[int]) -> Expr:
     """Creates an expression that references one or more columns by their index(es).
 
     Notes:
-        `nth` is not supported for Polars version<1.0.0. Please use [`col`](/api-reference/narwhals/#narwhals.col) instead.
+        `nth` is not supported for Polars version<1.0.0. Please use [`col`](../narwhals/#narwhals.col) instead.
```
Comment (FBruzzesi) [inline narwhals/expr.py:6090]: This was not working anyway, locally the new one does

**Review by MarcoGorelli — APPROVED** [review]
thanks @FBruzzesi ! happy to just deploy and see what happens

**Comment by FBruzzesi** [comment]
> thanks @FBruzzesi ! happy to just deploy and see what happens

It works 👌🚀

File: narwhals/dataframe.py [hunk 2]
```diff
@@ -337,12 +337,28 @@ def __eq__(self, other: object) -> NoReturn:
...
-    This class is not meant to be instantiated directly - instead, use
-    `narwhals.from_native`.
+    """Narwhals DataFrame, backed by a native eager dataframe.
+
+    !!! warning
+        This class is not meant to be instantiated directly - instead:
+
+        - If the native object is a eager dataframe from one of the supported
+            backend (e.g. pandas.DataFrame, polars.DataFrame, pyarrow.Table),
+            you can use [`narwhals.from_native`](../narwhals/#narwhals.from_native):
```
[the remark this answers is not in the document]
Comment (MarcoGorelli) [reply narwhals/dataframe.py:347]: looks like it works well, nice one

**Pull request merged**